[Monorepo] import gutenberg-mobile as a package

dratwas opened the pull request:
## Description

Explores #11491.
Superseeds #17456 started by @Tug 


Wordpress-iOS PR - https://github.com/wordpress-mobile/WordPress-iOS/pull/12762
Wordpress-Android PR - https://github.com/wordpress-mobile/WordPress-Android/pull/10668

This is a migration of gutenberg-mobile to gutenberg repository.

New packages added:

- `@wordpress/react-native-editor` - playground react-native app and bundle entrypoint
- `@wordpress/react-native-bridge` - module which starts react-native inside the client app (`WP-iOS/Android`)
- `@wordpress/react-native-aztec` - react-native-aztec module

Usual `yarn` commands ran from gutenberg-mobile are now accessible from gutenberg using `npm run native`

For instance:

`yarn start` => `npm run native start`

<!-- Please describe what you have changed or added -->

## How has this been tested?
TBD

## Screenshots <!-- if applicable -->

## Types of changes
Add new native packages, for native build, unifying the 2 repositories

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [ ] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [ ] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [ ] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've imported git history
- [ ] I've checked that the licencing is OK
- [ ]  I've checked that package release is working as expected (lerna)
- [x]  I've checked that installing dependencies works as expected and that changes to package-lock.json are OK
- [ ] I've checked that we are up to date with gutenberg-mobile develop branch
- [ ] I've informed concerned developers of the change of repository
- [x] I've ported e2e tests iOS
- [x] I've ported e2e tests Android
- [x] I've ported pre-commit hooks including eslint
- [x] I've ported i18n
- [x] I've checked that build scripts are up to date with WordPress-iOS
- [x] I've checked that build scripts are up to date with WordPress-Android
- [x] I've checked that WordPress-iOS works properly after migration
- [x] I've checked that WordPress-Android works properly after migration
- [x] I've made sure that the project can be opened in Android Studio
- [x] I've made sure that the project can be opened in XCode
- [x] I've made sure that the demo app for mobile runs
- [ ] I've made sure that I have excluded any binary files, jar, phar... from the patch (gutenberg-mobile has gradle-wrapper.jar committed to the repo though and we might want to fix that later?)
- [ ] I've checked that all CI steps are imported
- [ ] i've checked that the `react-native-aztec` example app works

pinarol committed:
[iOS]Fix: Oddly formatted text hangs Gutenberg (#1352)

* Enable batch editing on font change

* Remove unnecessary `removeAttribute` call

etoledom committed:
Update gutenberg ref

etoledom committed:
Adding iOS side of new `gutenbergDidAutosave` bridge event

daniloercoli committed:
Update GB hash

daniloercoli committed:
Update GB ref

SergioEstevao committed:
Intercept paste without formatting command.

mchowning committed:
Merge pull request #1358 from wordpress-mobile/release/1.12.0

Release v1.12.0

mchowning committed:
Merge pull request #1359 from wordpress-mobile/master

Merge master (v1.12.0) back to develop

geriux committed:
Update Gutenberg ref

Tug committed:
Fix store crash with gutenberg master

Tug committed:
Update gutenberg ref

Tug committed:
Update gutenberg ref

SergioEstevao committed:
Update Gutenberg version.

daniloercoli committed:
Update GB hash after merging the companion PR

daniloercoli committed:
Update bundle

daniloercoli committed:
Merge branch 'develop' of https://github.com/wordpress-mobile/gutenberg-mobile into add/autosave-monitor

* 'develop' of https://github.com/wordpress-mobile/gutenberg-mobile:
  [iOS]Fix: Oddly formatted text hangs Gutenberg (#1352)
  Remove redundant bg color within button appender (#1348)
  Update bundles
  Update package.json version to 1.12.0

# Conflicts:
#	bundle/android/App.js
#	bundle/android/App.js.map
#	bundle/ios/App.js
#	bundle/ios/App.js.map

daniloercoli committed:
Update bundles

daniloercoli committed:
Merge pull request #1351 from wordpress-mobile/add/autosave-monitor

Add "Autosave" bridge methods and enable local auto-save of posts

SergioEstevao committed:
Merge branch 'develop' into issue/562_add_underline

# Conflicts:
#	gutenberg

SergioEstevao committed:
Update GB reference.

mchowning committed:
Update AztecEditor-iOS to 1.9.0

mchowning committed:
Bump version to 1.13.0

mchowning committed:
Update bundles

Tug committed:
Remove unused require

Tug committed:
Fix tests

Tug committed:
Merge remote-tracking branch 'origin/develop' into update/fix-store-crash

Tug committed:
Add missing @wordpress/block-directory package in symlinked-packages

Tug committed:
Update gutenberg ref

SergioEstevao committed:
Only report selection changes if text view is still first responder.

SergioEstevao committed:
Don't allow interactions with URL in Aztec text blocks.

daniloercoli committed:
Merge pull request #1372 from wordpress-mobile/release/1.13.0

Release/1.13.0

daniloercoli committed:
Merge pull request #1376 from wordpress-mobile/master

Merge master (v1.13.0) back to develop

geriux committed:
Updated gutenberg ref

geriux committed:
Merge branch 'develop' of github.com:wordpress-mobile/gutenberg-mobile into feature/move-media-upload-progress

SergioEstevao committed:
Update GB reference.

SergioEstevao committed:
Update GB reference.

geriux committed:
Pass media filter on media upload

geriux committed:
Update gutenberg ref

Tug committed:
Use real timers for e2e tests

SergioEstevao committed:
Update GB reference.

SergioEstevao committed:
Move selection to the start of the URL.

SergioEstevao committed:
Make sure links selection is still possible.

SergioEstevao committed:
Work around for link selection.

SergioEstevao committed:
Join two guards.

SergioEstevao committed:
Update GB reference.

Tug committed:
Use fake timers in unit tests not config

daniloercoli committed:
Enable the CSSUnderlinePlugin on Aztec Android

daniloercoli committed:
Make sure to handle the new available format `underline`, passed from the JS side to Native when Underline is enabled on the current selection

daniloercoli committed:
Update GB ref

SergioEstevao committed:
Update GB reference.

SergioEstevao committed:
Merge pull request #1374 from wordpress-mobile/issue/1368_add_link_to_selected_text

Only report selection changes if text view is still first responder.

SergioEstevao committed:
Update package version.

SergioEstevao committed:
Merge branch 'master' into hotfix/1.11.1

# Conflicts:
#	package.json

SergioEstevao committed:
Update GB reference.

SergioEstevao committed:
Merge branch 'develop' into issue/1373_tap_links_opens_safari

SergioEstevao committed:
Merge pull request #1382 from wordpress-mobile/hotfix/1.11.1

Work around for link selection.

SergioEstevao committed:
Update GB reference.

SergioEstevao committed:
Update GB reference.

SergioEstevao committed:
Merge pull request #1377 from wordpress-mobile/issue/1261_add_missing_header_options

Add missing heading options

SergioEstevao committed:
Merge branch 'develop' into issue/1311_fix_paste_without_formatting

SergioEstevao committed:
Update GB reference.

SergioEstevao committed:
Merge branch 'develop' into issue/562_add_underline

geriux committed:
Update gutenberg ref

geriux committed:
Update gutenberg ref

geriux committed:
Update gutenberg ref

Tug committed:
Update gutenberg ref

Tug committed:
Update gutenberg ref

SergioEstevao committed:
Update GB reference.

Tug committed:
Merge remote-tracking branch 'origin/develop' into update/fix-store-crash

Tug committed:
Add comment explaining why we need to call jest.runAllTicks();

lukewalczak committed:
Fix lint issue on rmobile/master on gutenberg repo (#1388)

* Add ref to gutenberg repo

* Update ref and revert gitmodules

SergioEstevao committed:
Update GB reference.

Tug committed:
Merge remote-tracking branch 'origin/develop' into update/fix-store-crash

SergioEstevao committed:
Merge branch 'develop' into issue/1311_fix_paste_without_formatting

Tug committed:
Update gutenberg ref

SergioEstevao committed:
Merge pull request #1360 from wordpress-mobile/issue/1311_fix_paste_without_formatting

Intercept paste without formatting command.

SergioEstevao committed:
Update GB reference.

SergioEstevao committed:
Merge branch 'develop' into issue/562_add_underline

etoledom committed:
Update gutenberg ref

Tug committed:
Merge remote-tracking branch 'origin/develop' into update/fix-store-crash

geriux committed:
Update gutenberg ref

geriux committed:
Update gutenberg ref

geriux committed:
Merge branch 'develop' of github.com:wordpress-mobile/gutenberg-mobile into feature/move-media-upload-progress

# Conflicts:
#	gutenberg

geriux committed:
Gitmodules update

geriux committed:
Update gutenberg ref

SergioEstevao committed:
Update GB reference.

koke committed:
Merge pull request #1361 from wordpress-mobile/feature/move-media-upload-progress

Move MediaUploadPorgress to its own component folder

SergioEstevao committed:
Update GB reference.

Tug committed:
Increase the number of workers so tests don't get queued when calling driver.init to saucelabs

Tug committed:
Reenable disabled e2e tests

Tug committed:
Reenable disabled e2e tests

Tug committed:
Update gutenberg ref

Tug committed:
Update gutenberg ref

Tug committed:
Disable warning notices

Tug committed:
Disable warning notices, fix lint

Tug committed:
Merge remote-tracking branch 'origin/develop' into update/fix-store-crash

SergioEstevao committed:
Update GB reference.

SergioEstevao committed:
Merge branch 'develop' into issue/1373_tap_links_opens_safari

Tug committed:
Update gutenberg ref

SergioEstevao committed:
Update GB reference.

Tug committed:
Revert max workers to 3 and increase timeout in tests for now

Tug committed:
Update gutenberg ref

Tug committed:
Merge pull request #1366 from wordpress-mobile/update/fix-store-crash

Fix store crash with gutenberg master

SergioEstevao committed:
Update GB reference.

lukewalczak committed:
Add isAppender functionality to MediaPlaceholder (#1327)

* set ref of gutenberg to PR-17195

SergioEstevao committed:
Only apply selection fix on iOS 13.0.0

SergioEstevao committed:
Merge branch 'develop' into issue/1373_tap_links_opens_safari

# Conflicts:
#	gutenberg

SergioEstevao committed:
Update GB reference.

jbinda committed:
Feat: Cross platform InspectorControls (#1341)

* update ref to rnmobile/inpector-controls

SergioEstevao committed:
Update release notes.

SergioEstevao committed:
Update GB reference.

geriux committed:
iOS bridge updated with media type

SergioEstevao committed:
Update GB reference.

SergioEstevao committed:
Update GB reference.

etoledom committed:
Update gutenberg ref

SergioEstevao committed:
Update GB reference.

SergioEstevao committed:
Merge pull request #1393 from wordpress-mobile/master

Merge master (v1.13.1) back to develop

SergioEstevao committed:
Merge branch 'develop' into issue/1373_tap_links_opens_safari

# Conflicts:
#	react-native-aztec/ios/RNTAztecView/RCTAztecView.swift

etoledom committed:
Update gutenberg ref

SergioEstevao committed:
Merge pull request #1375 from wordpress-mobile/issue/1373_tap_links_opens_safari

Don't allow interactions with URL in Aztec text blocks.

etoledom committed:
Update gutenberg ref

geriux committed:
Mock media type for media upload

geriux committed:
Update gutenberg ref

geriux committed:
Return media type in Android's bridge

jbinda committed:
Refactor: BottomSheet controls (#1380)

* update ref to master

marecar3 committed:
Fix when multiple media selection adds only one image or video block (#1383)

dratwas committed:
Feat: support of multiple selection in media upload (#1331)

* wip - ios multiple

* update gutenberg ref

* update bundle ios

* update ref

* update ref

* update example app

* fix methods parameters description and use constants for dictionary keys

* add android support

* update js bundle

* add RNMedia interface and use it

* add multiple upload functionality for android

* support multiple selection of video from device

* update ref

* Update bundles

* update bundle and fix ids in tuple

* update ref

* update ref

* update bundle

* update js

etoledom committed:
Update gutenberg ref

etoledom committed:
Merge branch 'develop' into issue/1337-link-to-open-new-tab

etoledom committed:
Adding Link Target entry to release notes 1.15.0

koke committed:
Update Gutenberg

etoledom committed:
Update gutenberg ref to master

etoledom committed:
Merge remote-tracking branch 'origin/develop' into issue/darkmode-toolbar-buttons

etoledom committed:
Merge pull request #1357 from wordpress-mobile/issue/darkmode-toolbar-buttons

iOS DarkMode toolbar buttons

etoledom committed:
Update gutenberg ref to master

etoledom committed:
Merge remote-tracking branch 'origin/develop' into issue/1337-link-to-open-new-tab

etoledom committed:
Merge pull request #1394 from wordpress-mobile/issue/1337-link-to-open-new-tab

Link target option on Image Block

jbinda committed:
Feat: Ungrouping (#1381)

* update ref to master

lukewalczak committed:
Add FloatingToolbar component (#1362)

* Add ref to gutenberg repo

* Update ref

* Update ref

* Update ref

* Update ref and add style variable

* Update ref

* Update ref

* Update ref

geriux committed:
Merge branch 'develop' of github.com:wordpress-mobile/gutenberg-mobile into feature/media-text-improvements

# Conflicts:
#	gutenberg
#	ios/gutenberg/GutenbergViewController.swift
#	react-native-gutenberg-bridge/android/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/GutenbergBridgeJS2Parent.java
#	react-native-gutenberg-bridge/android/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/RNReactNativeGutenbergBridgeModule.java
#	react-native-gutenberg-bridge/android/src/main/java/org/wordpress/mobile/WPAndroidGlue/WPAndroidGlueCode.java
#	react-native-gutenberg-bridge/ios/GutenbergBridgeDelegate.swift
#	react-native-gutenberg-bridge/ios/RNReactNativeGutenbergBridge.swift

geriux committed:
Update to fork

SergioEstevao committed:
Merge branch 'develop' into issue/562_add_underline

# Conflicts:
#	gutenberg

SergioEstevao committed:
Update Aztec to version 1.10.1

SergioEstevao committed:
Merge pull request #1370 from wordpress-mobile/issue/562_add_underline

Issue/562 add underline

marecar3 committed:
Release/1.14.0 (#1392)

marecar3 committed:
Merge branch 'develop' into master-v1.14.0-to-develop

# Conflicts:
#	bundle/android/App.js
#	bundle/android/App.js.map
#	bundle/ios/App.js
#	bundle/ios/App.js.map
#	gutenberg
#	react-native-gutenberg-bridge/android/src/main/java/org/wordpress/mobile/WPAndroidGlue/WPAndroidGlueCode.java

marecar3 committed:
Revert gb ref to latest from develop

geriux committed:
Media & Text - Android: Fix return media type

geriux committed:
Update gutenberg ref

mkevins committed:
Add upload media to button listener interface in glue code

mkevins committed:
Add "media" MediaType

mkevins committed:
Detect "media" MediaType from filter

mkevins committed:
Invoke button listener method for "media" MediaType

mkevins committed:
Use MediaType static import consistently

etoledom committed:
[RNAztec iOS]: Improve sync of placeholder attributes and typing attributes

etoledom committed:
Update gutenberg ref

etoledom committed:
Update release notes

geriux committed:
Update gutenberg ref

jbinda committed:
Feat: Navigation Upward in InnerBlocks (#1371)

* Add ref to gutenberg repo with navigation upward in inner block

etoledom committed:
Update gutenberg ref

etoledom committed:
Merge remote-tracking branch 'origin/develop' into master-v1.14.0-to-develop

etoledom committed:
Update JS Bundles

etoledom committed:
Merge pull request #1410 from wordpress-mobile/master-v1.14.0-to-develop

Merge master (v1.14.0) back to develop

etoledom committed:
Merge remote-tracking branch 'origin/develop' into issue/1391-ios-dark-mode-issues

etoledom committed:
Update gutenberg ref

etoledom committed:
Merge pull request #1415 from wordpress-mobile/issue/1391-ios-dark-mode-issues

iOS dark mode improvements

koke committed:
Update Gutenberg

koke committed:
Merge branch 'develop' into media-text-stacking

koke committed:
Update Gutenberg

SergioEstevao committed:
Update to iOS11 and Swift 5.

SergioEstevao committed:
Remove check for iOS 11

lukewalczak committed:
Fix SafeArea (#1425)

* Add ref to gutenberg repo

* Revert changes in gitmodules and update ref

koke committed:
Update Gutenberg

koke committed:
Merge branch 'develop' into media-text-stacking

koke committed:
Merge pull request #1400 from wordpress-mobile/media-text-stacking

Make media-text respect stacking setting on native

SergioEstevao committed:
Merge pull request #1426 from wordpress-mobile/issue/update_to_iOS11_Xcode11

Update to iOS11 and Swift 5.

koke committed:
Update Gutenberg

koke committed:
Merge branch 'develop' into feature/media-text-improvements

koke committed:
Update gutenberg submodule

koke committed:
Update gutenberg submodule

koke committed:
Updated bundles

koke committed:
Merge pull request #1378 from wordpress-mobile/feature/media-text-improvements

Media & Text - Media picker buttons functionality

mkevins committed:
Merge branch 'develop' into try/fix-media-text-device-picker-android

mkevins committed:
Update gutenberg reference

mkevins committed:
Call media library method for media mediatype

koke committed:
Update deployment target for RNTAztecView to iOS 11

koke committed:
Fix header search path for libmxl2

etoledom committed:
Bump swift_version to 5.0 on RNTAztecView podspec

koke committed:
Merge pull request #1434 from wordpress-mobile/rntaztec-ios11

Update RNTAztec to iOS11

koke committed:
Import new location for base styles

koke committed:
Update gutenberg

koke committed:
Update gutenberg

koke committed:
Update bundles

koke committed:
Update release notes

lukewalczak committed:
Don't display UngroupingButton for Media&Text block (#1420)

* Add ref to gutenberg repo

* Update ref

* Update ref

* Revert changes in gitmodules and update ref

koke committed:
Update gutenberg

koke committed:
Update gutenberg

koke committed:
Update gutenberg

koke committed:
Merge branch 'develop' into issue/fix-scrollViewRef-crash

koke committed:
Merge pull request #1438 from wordpress-mobile/issue/fix-scrollViewRef-crash

Fix(?) crash when removing media-text blocks

mkevins committed:
Merge branch 'develop' into try/fix-media-text-device-picker-android

mkevins committed:
Merge pull request #1412 from wordpress-mobile/try/fix-media-text-device-picker-android

Filter for images and video in add media picker

hypest committed:
Merge pull request #1435 from wordpress-mobile/add/package-base-styles

Import new location for base styles

koke committed:
Update Gutenberg

koke committed:
Merge branch 'develop' into fix/button-active

koke committed:
Merge pull request #1440 from wordpress-mobile/fix/button-active

Fix Button active status

koke committed:
Update Gutenberg

koke committed:
Merge branch 'develop' into enable-media-text

koke committed:
Updated bundles

koke committed:
Merge pull request #1439 from wordpress-mobile/enable-media-text

Enable media-text in all builds

koke committed:
Update Gutenberg

koke committed:
Add Media & Text block to initial HTML

koke committed:
Update Gutenberg

koke committed:
Merge pull request #1443 from wordpress-mobile/issue/1411-media-text-icon-dark

Fix icons for Media & Text in dark mode

gziolo commented:
For reference, an old PR where @omarreiss moved https://github.com/WordPress/packages with all git history to Gutenberg: https://github.com/WordPress/gutenberg/pull/7556. I'm sure we should take the same approach here to respect all contributors.

The magic trick happened with:
`git subtree add --prefix temp-packages https://github.com/WordPress/packages.git master`

dratwas committed:
Add 'packages/react-native-editor/' from commit '3e9536a973321c9c7e31ad46fc8d36395251f7dc'

git-subtree-dir: packages/react-native-editor
git-subtree-mainline: 0ee2b784c10ace28f6c84d079ec601ff80a76199
git-subtree-split: 3e9536a973321c9c7e31ad46fc8d36395251f7dc

dratwas committed:
move packages to its directories

dratwas committed:
split to packages

dratwas committed:
add packages to package.json

dratwas committed:
fix tests

dratwas committed:
fix tests and lint

dratwas committed:
fix image test

dratwas committed:
fix import of aztec

dratwas committed:
update package-lock.json

dratwas committed:
remove makepot

dratwas committed:
working e2e ios temp

dratwas committed:
add project root

dratwas committed:
move podspecs and fix lint issues in aztec

dratwas committed:
move proptyopes

dratwas committed:
remove class  properties from aztec

dratwas committed:
experimental - move podspec to root lvl

dratwas committed:
fix podspec

dratwas committed:
fix paths

dratwas committed:
add bundle

dratwas committed:
add flow strip types plugin

dratwas committed:
support wp android script

dratwas committed:
move jitpack.yml

dratwas committed:
Merge branch 'master' into rnmobile/monorepo

dratwas committed:
add base styles

dratwas committed:
temporary fix for build

dratwas committed:
exclude device tests from unit tests

dratwas committed:
remove unused colors and fix lint

dratwas committed:
exclude native test from unit tests

Tug commented on webpack.config.js:
```diff
@@ -29,6 +29,11 @@ const WORDPRESS_NAMESPACE = '@wordpress/';
 	.filter( ( packageName ) => packageName.startsWith( WORDPRESS_NAMESPACE ) )
+	// TODO fix config to build them as well.
+	.filter( ( packageName ) =>
+		! ( packageName.includes( 'react-native-editor' ) ) &&
```
Could simply use `packageName.startsWith( 'react-native' )`.
Btw do you think we can have webpack build our native bundles? That would be awesome :)

dratwas replied:
>Could simply use packageName.startsWith( 'react-native' ).

Yeah, good idea :)

>Btw do you think we can have webpack build our native bundles? That would be awesome :)

This is what Haul does, so it is possible, but TBH i'm not sure if it should be a part of this PR :) WDYT?

Tug replied:
> This is what Haul does, so it is possible

Awesome :)

> but TBH i'm not sure if it should be a part of this PR :) WDYT?

Yeah probably not.
If we could have minimal changes in this PR that would be great

dratwas committed:
revert colors

dratwas committed:
try ignore react-native-editor in build script

dratwas committed:
tmp - fix lint-types

dratwas committed:
fix script to remove globals from r-n types

dratwas committed:
remove fox types from postinstall

dratwas committed:
change fix-types script

dratwas committed:
fix webpack config

dratwas committed:
trigger CI

dratwas commented:
MOVED TO https://github.com/WordPress/gutenberg/pull/18159 since we can not use a fork with Jitpack build.
